master/internal/api_project.go

```diff
@@ -185,6 +190,83 @@ func (a *apiServer) getProjectColumnsByID(
 	if err != nil {
 		return nil, err
 	}
+
+	trialIDs := make([]int, 0, len(hyperparameters)+1)
+	// dummy value so bun doesn't generate invalid sql
+	trialIDs = append(trialIDs, -1)
+	for _, hparam := range hyperparameters {
+		if hparam.BestTrialID != nil {
+			trialIDs = append(trialIDs, *hparam.BestTrialID)
+		}
+	}
+	summaryMetrics := []struct {
+		ID             int
+		SummaryMetrics model.JSONObj
+	}{}
+
+	trialsQuery := db.Bun().NewSelect().
+		Column("id").
+		Column("summary_metrics").
+		Table("trials").
+		Where("id IN (?)", bun.In(trialIDs)).
+		Order("id")
```

Comment:
I don't think you make any use of the order

```suggestion
		Where("id IN (?)", bun.In(trialIDs))
```

Reply:
Postgres doesn't guarantee return order and we want the columns to return with a stable order.

Reply:
shouldn't you sort the column names then? when you sort trials, and load summary metrics jsonb, and decode it into a go map, there's lots of risk the column names will get reordered.

Reply:
we sort the keys inside the summary metrics [here](https://github.com/determined-ai/determined/blob/3284e7a0acd0fff55c07a25588780d0f66365b0a/master/internal/api_project.go#L219-L225)

Reply:
how are the columns deduplicated? am I reading it right that if you have the same metric for, say, 10 experiments with 10 trials, the column will get added 10 times here: https://github.com/determined-ai/determined/blob/3284e7a0acd0fff55c07a25588780d0f66365b0a/master/internal/api_project.go#L257

Reply:
we have a map of keys we've previously seen [here](https://github.com/determined-ai/determined/pull/7320/files#diff-759959d3eba07f9028dad61e59aa27a4e2b7601f3424ead0036d17e7e2ce4cb5R216), and then check for membership and mark as seen [here](https://github.com/determined-ai/determined/pull/7320/files#diff-759959d3eba07f9028dad61e59aa27a4e2b7601f3424ead0036d17e7e2ce4cb5R228-R229)

Reply:
ok, thank you.

let's say we have trial id = 1 with metrics [loss, a1, b1]. trial id = 2 with metrics [loss, a2, b2]. 
then the output columns will be [a1, b1, loss, a2, b2].
if we swap them, and have trial id = 1 with metrics [loss, a2, b2], and trial id = 2 with metrics [loss, a1, b1],
then the output columns will be [a2, b2, loss, a1, b1].

is this the behavior we want? from user experience perspective, it'd be confusing for me as to why the columns appear in that order. wouldn't it be better to sort the columns alphabetically?

Reply:
For now, this is the behavior we want. This matches the behavior for validation metrics. If we get feedback expressing confusion as to the order of columns, we can change that.

Reply:
yep, I'd express confusion both from the UX perspective and from reading the code.